hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -630,6 +635,26 @@ void resetMigrationInterceptor() {
         migrationInterceptor = new MigrationInterceptor.NopMigrationInterceptor();
     }
 
+    boolean onDemoteRequest(Member member) {
+        if (!partitionStateManager.isInitialized()) {
+            if (demoteRequestedMembers.add(member)) {
+                logger.info("Demote request of " + member + " is handled");
+            }
+            sendDemoteResponseOperation(member);
```

Comment:
Is it intentional for this call to be outside the `demoteRequestedMembers.add(member)` conditional check? The operation seems idempotent to me, but I wanted to flag so we can make sure!

Reply:
It is intentional to solve cases where the first `DemoteResponseOperation` is never received on the other end for some reason and a new `DemoteRequestOperation` is sent as a retry.

I updated `DemoteResponseOperation`/ `com.hazelcast.internal.partition.impl.InternalPartitionServiceImpl#onDemoteResponse` so that it is also idempotent.